pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -103,6 +107,10 @@ func (r *Cluster) ValidateUpdate(old runtime.Object) error {
 
 	allErrs = append(allErrs, ValidateWorkerKubernetesVersionSkew(r, oldCluster)...)
 
+	if err := validateEtcdEncryptionConfig(r.Spec.EtcdEncryption); err != nil {
```

Comment:
do we need these validations in the CLI as well?

Reply:
yeah, I will add in this and more validations the CLI as well. I was thinking I would do it in a follow-up PR when I make other CLI changes. Does that sound good?